### PR TITLE
Fix nclx color profile reading introduced in libheif 1.10.0

### LINKED ISSUE #49: Since libheif 1.10.0 cannot open non-rICC color profile image
From my little investigation, I found that since [libheif 1.10.0](https://github.com/strukturag/libheif/blob/v1.10.0/libheif/heif.cc#L1139) the `heif_image_handle_get_raw_color_profile_size` always returns 0 for non-rICC color profile images.

I suspect this leads to the following error when opening a nclx image:

```
HeifError                                 Traceback (most recent call last)
<ipython-input-15-49240deda0a0> in <module>
----> 1 pyheif.read("./0_1 (1).heic")

~/.local/lib/python3.9/site-packages/pyheif/reader.py in read(fp, apply_transformations, convert_hdr_to_8bit)
     35 def read(fp, *, apply_transformations=True, convert_hdr_to_8bit=True):
     36     d = _get_bytes(fp)
---> 37     result = _read_heif_bytes(d, apply_transformations, convert_hdr_to_8bit)
     38     return result
     39

~/.local/lib/python3.9/site-packages/pyheif/reader.py in _read_heif_bytes(d, apply_transformations, convert_hdr_to_8bit)
     70     ctx = _libheif_cffi.lib.heif_context_alloc()
     71     try:
---> 72         result = _read_heif_context(ctx, d, apply_transformations, convert_hdr_to_8bit)
     73     finally:
     74         _libheif_cffi.lib.heif_context_free(ctx)

~/.local/lib/python3.9/site-packages/pyheif/reader.py in _read_heif_context(ctx, d, apply_transformations, convert_hdr_to_8bit)
     98
     99     try:
--> 100         result = _read_heif_handle(handle, apply_transformations, convert_hdr_to_8bit)
    101     finally:
    102         _libheif_cffi.lib.heif_image_handle_release(handle)

~/.local/lib/python3.9/site-packages/pyheif/reader.py in _read_heif_handle(handle, apply_transformations, convert_hdr_to_8bit)
    146
    147     metadata = _read_metadata(handle)
--> 148     color_profile = _read_color_profile(handle)
    149
    150     heif_file = HeifFile(

~/.local/lib/python3.9/site-packages/pyheif/reader.py in _read_color_profile(handle)
    220     error = _libheif_cffi.lib.heif_image_handle_get_raw_color_profile(handle, p_data)
    221     if error.code != 0:
--> 222         raise _error.HeifError(
    223             code=error.code,
    224             subcode=error.subcode,

HeifError: Code: 10, Subcode: 0, Message: "Color profile does not exist: Unspecified"
```

### PR DESCRIPTION
**Pull request by mskrip**
**Disclaimer**:
Please beware that I am by no means an expert at libheif or cffi bindings in general and I only fixed the code so that the included tests pass with `libheif@1.12.0`as I had some trouble opening some images using the older versions but which work on this one.

Close #49

### CODE REVIEW

File: pyheif/reader.py [hunk 1]
```diff
@@ -205,13 +205,26 @@ def _read_color_profile(handle):
...
-    elif profile_type == _constants.heif_color_profile_type_rICC:
-        color_profile["type"] = "rICC"
-    elif profile_type == _constants.heif_color_profile_type_prof:
-        color_profile["type"] = "prof"
-    data_length = _libheif_cffi.lib.heif_image_handle_get_raw_color_profile_size(handle)
-    p_data = _libheif_cffi.ffi.new("char[]", data_length)
-    error = _libheif_cffi.lib.heif_image_handle_get_raw_color_profile(handle, p_data)
+        data_length = _libheif_cffi.ffi.sizeof("struct heif_color_profile_nclx")
+        p_p_data = _libheif_cffi.ffi.new("struct heif_color_profile_nclx * *")
+        error = _libheif_cffi.lib.heif_image_handle_get_nclx_color_profile(
```
[the remark this answers is not in the document]
Comment (homm) [reply pyheif/reader.py:210]: @mskrip will you merge it?

**Comment by ant32bit-carsales** [comment]
@mskrip if you have an image that failed to work previously, it would be great to add it to the test suite so we can ensure that new updates don't break this again.

**Comment by homm** [comment]
Those images right in test suite right now, they are `93FG5559.HIF` and `93FG5564.HIF`.

**Comment by mskrip** [comment]
Yes, the existing tests already failed for me on the newer libheif version.

**Comment by ant32bit-carsales** [comment]
OK got it, it wasn't failing for me because the test were run on libheif 1.7.0 in docker. This change requires libheif >=1.10.0 so the docker fails to build but I'm going to merge it anyway, as there's another pull request #63 which updates libheif to 1.12.0 and fixes the issue.

**Pull request merged**